Jenkinsfile

```diff
@@ -13,45 +13,31 @@ pipeline {
   }
 
   stages {
+
+    stage("Build Docker Image"){
+        steps { sh './build.sh' }
+    }
+
     stage('Validate Changelog'){
         steps { sh './validate.sh' }
     }
 
     stage('Test') {
       steps {
         sh './test.sh'
-        junit 'rspec_junit.xml'
+      }
+      post {
+          always{
+            junit 'rspec_junit.xml'
+          }
       }
     }
 
     // Only publish to RubyGems if branch is 'master'
     // AND someone confirms this stage within 5 minutes
-    stage('Publish to RubyGems?') {
+    stage('Conditional publish to RubyGems and Dockerhub') {
       agent { label 'releaser-v2' }
-
-      when {
-        allOf {
-          branch 'master'
-          expression {
-            boolean publish = false
-
-            if (env.PUBLISH_GEM == "true") {
-                return true
-            }
-
-            try {
-              timeout(time: 5, unit: 'MINUTES') {
-                input(message: 'Publish to RubyGems?')
-                publish = true
-              }
-            } catch (final ignore) {
-              publish = false
-            }
-
-            return publish
-          }
-        }
-      }
+      when { tag "v*" }
```

Comment:
💥

Reply:
maybe it's worth adding a comment here that in this case there is a `TAG_NAME` environment variable that is set (as described below)?